node/cn/api_sc_backend_test.go

```diff
@@ -0,0 +1,402 @@
+
+	assert.Equal(t, expectedTx, tx)
+	assert.Equal(t, expectedHash, hash)
+	assert.Equal(t, expectedBlockNum, blockNum)
+	assert.Equal(t, expectedIndex, index)
+	assert.Equal(t, expectedReceipt, receipt)
+}
+
+func TestServiceChainAPIBackend_GetTxAndLookupInfoInCache(t *testing.T) {
+	api, mockCtrl, mockBlockChain, _ := prepareServiceChainAPIBackendTest(t)
+	defer mockCtrl.Finish()
+
+	expectedTx := tx1
+	expectedHash := hashes[0]
+	expectedBlockNum := uint64(123)
+	expectedIndex := uint64(321)
+
+	mockBlockChain.EXPECT().GetTxAndLookupInfoInCache(hashes[0]).Return(tx1, hashes[0], uint64(123), uint64(321)).Times(1)
+
+	tx, hash, blockNum, index := api.GetTxAndLookupInfoInCache(hashes[0])
+
+	assert.Equal(t, expectedTx, tx)
+	assert.Equal(t, expectedHash, hash)
+	assert.Equal(t, expectedBlockNum, blockNum)
+	assert.Equal(t, expectedIndex, index)
+}
+
+func TestServiceChainAPIBackend_GetBlockReceiptsInCache(t *testing.T) {
+	api, mockCtrl, mockBlockChain, _ := prepareServiceChainAPIBackendTest(t)
+	defer mockCtrl.Finish()
+
+	rct1 := newReceipt(111111)
+	rct2 := newReceipt(222222)
+	rct3 := newReceipt(333333)
+
+	receipts := types.Receipts{rct1, rct2, rct3}
+
+	mockBlockChain.EXPECT().GetBlockReceiptsInCache(hashes[0]).Return(receipts).Times(1)
+	returnedReceipts := api.GetBlockReceiptsInCache(hashes[0])
+
+	assert.Equal(t, receipts, returnedReceipts)
+}
+
+func TestServiceChainAPIBackend_GetTxLookupInfoAndReceiptInCache(t *testing.T) {
+	api, mockCtrl, mockBlockChain, _ := prepareServiceChainAPIBackendTest(t)
+	defer mockCtrl.Finish()
+
+	rct := newReceipt(250000)
+
+	expectedTx := tx1
+	expectedHash := hashes[0]
+	expectedBlockNum := uint64(123)
+	expectedIndex := uint64(321)
+	expectedReceipt := rct
+
+	mockBlockChain.EXPECT().GetTxLookupInfoAndReceiptInCache(hashes[0]).Return(tx1, hashes[0], uint64(123), uint64(321), rct).Times(1)
+
+	tx, hash, blockNum, index, receipt := api.GetTxLookupInfoAndReceiptInCache(hashes[0])
+
+	assert.Equal(t, expectedTx, tx)
+	assert.Equal(t, expectedHash, hash)
+	assert.Equal(t, expectedBlockNum, blockNum)
+	assert.Equal(t, expectedIndex, index)
+	assert.Equal(t, expectedReceipt, receipt)
+}
+
+func TestServiceChainAPIBackend_ChainConfig(t *testing.T) {
+	api, mockCtrl, _, _ := prepareServiceChainAPIBackendTest(t)
+	defer mockCtrl.Finish()
+
+	assert.Equal(t, cc, api.ChainConfig())
+}
+
+func TestServiceChainAPIBackend_CurrentBlock(t *testing.T) {
+	api, mockCtrl, mockBlockChain, _ := prepareServiceChainAPIBackendTest(t)
+	defer mockCtrl.Finish()
+
+	mockBlockChain.EXPECT().CurrentBlock().Return(blocks[0]).AnyTimes()
+	assert.Equal(t, blocks[0], api.CurrentBlock())
+}
+
+func TestServiceChainAPIBackEnd_SetHead(t *testing.T) {
+	api, mockCtrl, mockBlockChain, _ := prepareServiceChainAPIBackendTest(t)
+	defer mockCtrl.Finish()
+
+	headNumber := uint64(12345)
+	mockBlockChain.EXPECT().SetHead(headNumber).Times(1)
+	api.SetHead(headNumber)
+}
+
```

Comment:
Thanks. @ehnuje. :)
Will you add unit test of `HeaderByNumber`, `BlockByNumber`, and  `StateAndHeaderByNumber` in next PR?

Reply:
@KimKyungup Of course. Tackling easier ones first =)